### PR TITLE
[Kimi-K3][ROCm] Fuse KDA upper-triangle zeroing

### PR DESCRIPTION
**Pull request by LiuYinfeng01**
Avoid a separate full `Akk` buffer memset by writing the required zeros in the existing fused solve kernel.

## Purpose

Kimi-K3's ROCm KDA prefill path allocates `Akk` with `torch.zeros`, even though the fused inter/solve Triton kernel already writes its lower triangle.

This change makes the fused kernel explicitly write the required upper-triangle zeros, allowing `Akk` to use `torch.empty` and avoiding a separate full-buffer memset.

It supports:

- Fixed-length and variable-length inputs
- Safe-gate and standard-gate paths
- Partial chunks with boundary masking

No documentation update is required because this is an internal performance optimization with no API or model-support changes.

## Test Plan

Run the targeted ROCm tests:

    python3 -m pytest -q tests/models/kimi_k3/test_amd_kda.py

Run Ruff checks:

    ruff check \
      vllm/models/kimi_k3/amd/ops/third_party/kda/chunk_intra.py \
      tests/models/kimi_k3/test_amd_kda.py

    ruff format --check \
      vllm/models/kimi_k3/amd/ops/third_party/kda/chunk_intra.py \
      tests/models/kimi_k3/test_amd_kda.py

Validate performance using Kimi-K3 TP8 on 8 × MI355X:

- Input length: 100,000 tokens
- Output length: 20 tokens
- Concurrency: 1
- Prefix caching: disabled
- Five measured requests per run
- Two independent baseline and patched runs

### Correctness

Targeted ROCm tests:

    4 passed

The tests cover:

- Fixed-length input with `safe_gate=False`
- Fixed-length input with `safe_gate=True`
- Variable-length input with `safe_gate=False`
- Variable-length input with `safe_gate=True`

Additional A/B validation confirmed:

- `Akk` matched the original implementation bit-for-bit
- End-to-end KDA output matched bit-for-bit
- The upper triangle remained zero for every tested sequence and partial chunk
- Ruff lint and formatting checks passed

### Accuracy / Numerical Parity

The optimization does not change the KDA arithmetic. It only moves the required upper-triangle zero writes from a standalone `torch.zeros` initialization into the existing fused solve kernel.

A direct baseline-versus-patched kernel comparison was performed with identical inputs.

Results:

- `Akk` matched the original zero-initialized implementation bit-for-bit.
- The complete KDA output matched bit-for-bit.
- Sequence lengths tested: `64`, `65`, `127`, `128`, `129`, and `1024`.
- Partial chunks and chunk-boundary cases were included.
- Fixed-length and variable-length layouts were tested.
- Both `safe_gate=True` and `safe_gate=False` produced an exactly zero upper triangle.
- No NaNs or numerical regressions were observed.
- All TP8 serving requests completed successfully.

The 100K serving environment was not deterministic at the generated-text level: repeated runs of the same patched server also produced different generated text. Therefore, generated-text equality was not used as numerical-parity evidence. Kernel-level tensor comparison was used instead and showed bit-for-bit output parity.
### Kernel Performance

Same-GPU alternating A/B using Kimi-K3 TP8 per-rank shapes:

    Tokens   Baseline median   Patched median   Improvement
    8,192    0.1614 ms         0.1580 ms        2.08%
    32,768   0.5193 ms         0.5157 ms        0.70%
    100,000  1.5398 ms         1.5250 ms        0.96%

### End-to-End Performance

Kimi-K3 TP8, 100K input, concurrency 1:

    Baseline mean TTFT, run 1: 7223.33 ms
    Baseline mean TTFT, run 2: 7221.91 ms
    Patched mean TTFT, run 1:  7206.20 ms
    Patched mean TTFT, run 2:  7210.31 ms

    Baseline average: 7222.62 ms
    Patched average:  7208.26 ms
    Improvement:      14.36 ms (0.20%)

All serving requests succeeded.

---
<details>
<summary>Essential Elements of an Effective PR Description Checklist</summary>

- [x] The purpose of the PR has been described.
- [x] The test plan includes the relevant test commands.
- [x] Correctness and performance results have been provided.
- [x] Documentation impact was considered; no update is required.

</details>

### CODE REVIEW

**Comment by LiuYinfeng01** [comment]
Hi @tjtanaa,

Could you please take a look at this PR when you have a chance? Your feedback would be greatly appreciated.

Thank you!

**Comment by hongxiayang** [comment]
@LiuYinfeng01 :

The improvement of 14.36 ms (0.20%) seems very minor — it's within run-to-run variance, especially since the two baseline runs already differ by ~1.4 ms. Without evidence of a more significant, reproducible gain, I'd lean toward shelving this PR.
